File: src/apps/kipo-platform/emisor/pdf_customization_constants.py
# Keep in sync with frontend: src/apps/kipo-dashboard/src/settings/ui/components/pdfCustomizationConstants.ts
ALLOWED_HTML_TAGS = frozenset({
    "h1", "h2", "h3", "h4", "h5", "h6", "div", "p", "span", "small", "br",
    "b", "i", "ul", "ol", "li", "strong", "table", "thead", "tbody", "tfoot", "tr", "th", "td",
})

DEFAULT_DISPLAY_OPTIONS: dict = {
    "show_catalog_codes": True,
    "show_product_key": True,
    "show_address_codes": True,
    "show_export_key": False,
    "round_unit_price": False,
    "show_tax_breakdown": True,
    "show_ieps_breakdown": True,
    "combine_ieps_with_subtotal": False,
    "repeat_signature_each_page": False,
}

KNOWN_DISPLAY_OPTION_KEYS = frozenset(DEFAULT_DISPLAY_OPTIONS.keys())

File: src/apps/kipo-dashboard/src/settings/ui/components/shared/pdfCustomizationConstants.ts
import type { UIInvoiceDisplayOptions } from './types'

// Keep in sync with backend: src/apps/kipo-platform/emisor/pdf_customization_constants.py
export const PDF_CUSTOM_SECTION_ALLOWED_TAGS = [
  'h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'div', 'p', 'span', 'small', 'br',
  'b', 'i', 'ul', 'ol', 'li', 'strong', 'table', 'thead', 'tbody', 'tfoot', 'tr', 'th', 'td',
] as const

export const DEFAULT_DISPLAY_OPTIONS: UIInvoiceDisplayOptions = {
  showCatalogCodes: true,
  showProductKey: true,
  showAddressCodes: true,
  showExportKey: false,
  roundUnitPrice: false,
  showTaxBreakdown: true,
  showIepsBreakdown: true,
  combineIepsWithSubtotal: false,
  repeatSignatureEachPage: false,
}
